pkg/render/goldmane/component.go

```diff
@@ -0,0 +1,320 @@
+					ServiceAccountName: GoldmaneServiceAccountName,
+					Tolerations:        tolerations,
+					ImagePullSecrets:   secret.GetReferenceList(c.cfg.PullSecrets),
+					Containers:         ctrs,
+					Volumes:            volumes,
+				},
+			},
+		},
+	}
+}
+
+func (c *Component) roleBinding() *rbacv1.RoleBinding {
+	return &rbacv1.RoleBinding{
+		TypeMeta: metav1.TypeMeta{Kind: "RoleBinding", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      GoldmaneRoleName,
+			Namespace: GoldmaneNamespace,
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "Role",
+			Name:     GoldmaneRoleName,
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      GoldmaneRoleName,
+				Namespace: GoldmaneNamespace,
+			},
+		},
+	}
+}
+
+func (c *Component) role() *rbacv1.Role {
+	policyRules := []rbacv1.PolicyRule{
+		{
+			APIGroups: []string{""},
+			Resources: []string{"configmaps"},
+			Verbs:     []string{"get", "list", "watch", "create", "update", "patch"},
+		},
+	}
+
+	return &rbacv1.Role{
+		TypeMeta: metav1.TypeMeta{Kind: "Role", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      GoldmaneRoleName,
+			Namespace: GoldmaneNamespace,
+		},
+		Rules: policyRules,
+	}
+}
+
+func (c *Component) deploymentSelector() *metav1.LabelSelector {
+	return &metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"app.kubernetes.io/name": GoldmaneDeploymentName,
+		},
+	}
+}
+
+func (c *Component) networkPolicy() *netv1.NetworkPolicy {
+	return &netv1.NetworkPolicy{
+		TypeMeta:   metav1.TypeMeta{Kind: "NetworkPolicy", APIVersion: "networking.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{Name: "allow-goldmane", Namespace: GoldmaneNamespace},
+		Spec: netv1.NetworkPolicySpec{
+			PodSelector: *c.deploymentSelector(),
+			PolicyTypes: []netv1.PolicyType{netv1.PolicyTypeIngress, netv1.PolicyTypeEgress},
+			Ingress: []netv1.NetworkPolicyIngressRule{
+				{
+					Ports: []netv1.NetworkPolicyPort{{
+						Protocol: ptr.ToPtr(corev1.ProtocolTCP),
+						Port:     ptr.ToPtr(intstr.FromInt32(GoldmaneServicePort)),
+					}},
+				},
+			},
+			Egress: []netv1.NetworkPolicyEgressRule{
+				{
+					To: []netv1.NetworkPolicyPeer{
+						{
+							PodSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{
+									"app.kubernetes.io/name": GuardianDeploymentName,
+								},
+							},
+							NamespaceSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{
+									"projectcalico.org/name": GuardianNamespace,
+								},
+							},
+						},
+					},
+					Ports: []netv1.NetworkPolicyPort{
+						{
+							Protocol: ptr.ToPtr(corev1.ProtocolTCP),
+							Port:     ptr.ToPtr(intstr.FromInt32(GuardianTargetPort)),
+						},
+					},
+				},
+				{
+					Ports: []netv1.NetworkPolicyPort{
+						{
+							Protocol: ptr.ToPtr(corev1.ProtocolUDP),
```

Comment:
In theory (but unlikely in practice) we could also see DNS traffic on TCP/53. Worth adding a rule for that

Reply:
Really? Is this for something other than DNS lookups?

I don't think we add DNS rules for tcp on 53 for other policies, unless I'm mistaken.

Reply:
Yep, we do actually do this for other policies as well.

https://github.com/tigera/operator/blob/cd83253b701673b4536c0d1190a134593cde67eb/pkg/render/common/networkpolicy/networkpolicy.go#L44-L79

Reply:
Looks like we only do it for OCP for some reason, but I think it applies to regular k8s DNS as well....

Reply:
Added it to the function I created for this.